waku/waku_archive/driver/sqlite_driver/migrations.nim

```diff
@@ -30,6 +51,15 @@ proc migrate*(db: SqliteDatabase, targetVersion = SchemaVersion): DatabaseResult
   ## NOTE: Down migration it is not currently supported
   debug "starting message store's sqlite database migration"
 
+  let userVersion = ? db.getUserVersion()
+  let numColumnsInPK = ? db.getNumColumnsInPK()
+
+  if userVersion == 0'i64 and numColumnsInPK == 3:
+    info "We found user_version 0 but the database schema reflects the user_version 7"
+    ## Force the correct schema version
+    ? db.setUserVersion( 7 )
+    return ok()
```

Comment:
What happens in case where

1. User is running v0.17.0
2. We update schema in v0.21.0
3. User updates to v0.21.0

IIUC the user_version gets set to `7` and then `migrate` proc returns and the migration does not happen - which is wrong - right? IMO the migration should continue with corrected `user_version`

Reply:
Good point!
In this case, the migration should not happen because the database is using the schema version 7 (three columns in PK) but the `user_version` value was 0.

If we update the schema in `0.21.0` we will need to revisit this logic of course. In fact, I think we need to modify this logic the next time we modify the schema somehow.